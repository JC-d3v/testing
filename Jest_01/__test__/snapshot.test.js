import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Pruebas de instantaneas ', () => {
  test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  test('Instantanea con variaciones date ', () => {
    const user = {
      createAt: new Date(),
      Id: Math.floor(Math.random() * 20),
    }
    expect(user).toMatchSnapshot();
  });

  test('Instantanea con excepcion en variaciones date ', () => {
    const user = {
      Id: Math.floor(Math.random() * 20),
      name: "Jorge C"
    }
    expect(user).toMatchSnapshot({
      Id: expect.any(Number)
    });
  });

});
