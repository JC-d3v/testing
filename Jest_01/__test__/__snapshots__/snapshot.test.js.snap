// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de instantaneas  Instantanea con Excepcion  1`] = `
Object {
  "Id": 15,
  "createAt": 2022-03-02T20:16:09.565Z,
}
`;

exports[`Pruebas de instantaneas  Instantanea con excepcion en variaciones date  1`] = `
Object {
  "Id": Any<Number>,
  "name": "Jorge C",
}
`;

exports[`Pruebas de instantaneas  Instantanea con variaciones date  1`] = `
Object {
  "Id": 10,
  "createAt": 2022-03-02T20:35:15.122Z,
}
`;

exports[`Pruebas de instantaneas  Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;
